tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_dummy_e2e_test.py

```diff
@@ -0,0 +1,164 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""E2E Tests for tfx.examples.chicago_taxi_pipeline.taxi_pipeline_beam."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import absl
+import os
+from typing import Text
+
+import tensorflow as tf
+
+from tfx.examples.chicago_taxi_pipeline import taxi_pipeline_beam
+from tfx.experimental.pipeline_testing import dummy_component_launcher
+from tfx.orchestration import metadata
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+from tfx.orchestration.config import pipeline_config
+
+class TaxiPipelineBeamEndToEndTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(TaxiPipelineBeamEndToEndTest, self).setUp()
+    self._test_dir = os.path.join(
+        os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()),
+        self._testMethodName)
+
+    self._pipeline_name = 'beam_test'
+    taxi_root = os.path.join(os.environ['HOME'],
+                             "tfx/tfx/examples/chicago_taxi_pipeline")
+    self._data_root = os.path.join(taxi_root, 'data', 'simple')
+    self._module_file = os.path.join(taxi_root, 'taxi_utils.py')
+    self._serving_model_dir = os.path.join(self._test_dir, 'serving_model')
+    self._pipeline_root = os.path.join(self._test_dir, 'tfx', 'pipelines',
+                                       self._pipeline_name)
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+
+  def assertExecutedOnce(self, component: Text) -> None:
+    """Check the component is executed exactly once.
+    Pipeline root is <test_dir>/tfx/pipelines/beam_test/
+    Execution outputs are saved in <component.id>/<key>/<execution.id>
+    """
+    component_path = os.path.join(self._pipeline_root, component)
+    self.assertTrue(tf.io.gfile.exists(component_path))
+    outputs = tf.io.gfile.listdir(component_path)
+    for output in outputs:
+      current_dir = os.path.join(component_path, output)
+      absl.logging.info("assertExecutedOnce: checking %s", current_dir)
+      execution = tf.io.gfile.listdir(current_dir)
+      self.assertLen(execution, 1)
+
+  def assertPipelineExecution(self) -> None:
```

Comment:
How about using component_ids defined in testTaxiPipelineBeam()? It looks redundant to list all components again. Or it might be better to call assertExecutedOnce directly in testTaxiPipelineBeam().